### DISCUSSION TITLE
Use the Middleware to log the start and the end of the request

### DISCUSSION THREAD

**Discussion by baymer** [vercel/next.js#46543] [Ideas]
### Describe the feature you'd like to request

Log information about request before it starts and after it'll be finished. For example:
- log opentelemetry meta (start span before, close and send span after);
- log to stdout with special format (like `{"message":"internal request completed POST /graphql query Selection 200 206ms", "request-id":"12345432343-12342412513",...}`);
- send metrics to graphana and others graphics, monitoring, alerts systems.

### Describe the solution you'd like

I don't known, may be like this:
```js
export function middleware(request: NextRequest) {
  console.log('request start');
  ...
  request.on('finish', (req, resp) => {
    console.log(`internal request completed ${req.method} ${req.path} ${resp.status} ${resp.duration}`);
  });
 ...
 return NextResponse.next();
}
```

### Describe alternatives you've considered

-

**Comment by rsweetland** [comment]
This discussion on [lifecycle event handlers](https://github.com/vercel/next.js/discussions/54831) I think is a good way to frame this feature.
